docs/application/native/guides/app-management/dali-widget-app.md

```diff
@@ -0,0 +1,268 @@
+
+The `mainloop()` function is used to start the event loop. If you do not call the function and start the event loop, DALi cannot call any callback functions for application events.
+
+The `RegisterWidgetCreatingFunction()` function registers creating function to create widget instance. Widget application will create widget instance using this creating function when WidgetViewer application requests.
+
+<a name="widget-api"></a>
+## Widget class
+
+`Dali::Widget` class provides interface for creating custom widget.
+
+The following table lists the main function to manage widget instance.
+
+**Table: Main functions**
+
+| Function           | Description                                                                   |
+|--------------------|-------------------------------------------------------------------------------|
+| `OnCreate()`       | Called after the widget instance is created.                                  |
+| `OnTerminate()`    | Called after the widget instance is terminated.                               |
+| `OnPause()`        | Called when the widget is invisible.                                          |
+| `OnResume()`       | Called when the widget is visible.                                            |
+| `OnResize()`       | Called before the widget size is changed.                                     |
+| `OnUpdate()`       | Called when an event for updating the widget is received.                     |
+| `SetContentInfo()` | Send data about current state of widget instance to WidgetViewer application. |
+
+Most of functions are for managing widget instance lifecycle. You can describe what will do on each lifecycle functions.
+
+Widget instance can send data to WidgetViewer application using `SetContentInfo()` function. If you want to save current state of widget before deleting and show it again, you can use it.
+
+<a name="create"></a>
+## Basics of Creating a DALi Widget Application
+
+1. Initialize the widget application:
+
+   Create the `Dali::WidgetApplication` class instance to initialize and set up DALi for a widget application
+   ```
+   int DALI_EXPORT_API main( int argc, char **argv )
+   {
+     WidgetApplication application = WidgetApplication::New( &argc, &argv, "" );
+
+     WidgetApplicationExample test( application );
+
+     application.MainLoop();
+
+     return 0;
+   }
+   ```
+
+2. Connect signals to keep yourself informed when certain system events occur.
+
+   To manage signal connection safely, DALi provides the [Dali::ConnectionTracker](../../api/mobile/latest/classDali_1_1ConnectionTracker.html) class. A typical way to start a DALi application is to create a class derived from the `Dali::ConnectionTracker` class and use its member functions as callback functions for DALi signals.When the application receives the `InitSignal()`, it can create UI components. Connect the `WidgetApplicationExample::Create()` callback to the `Dali::WidgetApplication::InitSignal()` function:
```

Comment:
Dali::ConnectionTracker> remove hyperlink

A typical way to start a DALi application is to create a class derived from the `Dali::ConnectionTracker` class and use its member functions as callback functions for DALi signals.When the application receives the `InitSignal()`, it can create UI components. Connect the `WidgetApplicationExample::Create()` callback to the `Dali::WidgetApplication::InitSignal()` function:

rephrase as:

A typical way to start a DALi application is to create a class derived from the `Dali::ConnectionTracker` class and use its member functions as callback functions for DALi signals. When the application receives the `InitSignal()` function, it can create user interface components. Additionally, connect the callback `WidgetApplicationExample::Create()` function to the ` Dali::WidgetApplication::InitSignal() ` function: